src/routes/common/filters/zod-error.filter.ts

```diff
@@ -0,0 +1,28 @@
+import {
+  ArgumentsHost,
+  Catch,
+  ExceptionFilter,
+  HttpStatus,
+} from '@nestjs/common';
+import { ZodError } from 'zod';
+import { Response } from 'express';
+
+/**
+ * This {@link ExceptionFilter} catches any {@link ZodError} thrown
+ * at the route level.
+ *
+ * It builds a JSON payload which contains a code and a message.
+ * The code and message used are read from {@link ZodError} issues.
+ */
+@Catch(ZodError)
+export class ZodErrorFilter implements ExceptionFilter {
```

Comment:
Nit: what about renaming this class and `ZodValidationPipe` to remove the `Zod` reference from the name? Ideally the caller would use a "validation" filter/pipe without knowing the underlying library implementing the validation checks. (No strong opinion)

Reply:
I've renamed the pipe to `ValidationPipe` but think we should keep the filter as is because of the error type. What do you think?

Reply:
Hmm, yes, makes sense as it is more specific and tied to the library 👍 Thanks!